src/client/serialize/daos_serialize.c

```diff
@@ -184,7 +184,7 @@ serialize_str(hid_t file_id, struct daos_prop_entry *entry, const char *prop_str
 		D_ERROR("failed to create attribute\n");
 		D_GOTO(out, rc = -DER_MISC);
 	}
-	status = H5Awrite(usr_attr, attr_dtype, entry->dpe_str);
+	status = H5Awrite(usr_attr, attr_dtype, entry->dpe_str ? entry->dpe_str : "");
```

Comment:
is it necessary to add the attribute if the label is empty? probably a question to @daltonbohning or @dsikich when she comes back.

Reply:
It's not strictly necessary, but I'm not sure which approach is better. Since container query lists properties that are empty, maybe serialization should also serialize properties that are empty? FWIW, this applies to any "string" container property, which currently is just DAOS_PROP_CO_LABEL, DAOS_PROP_CO_OWNER, DAOS_PROP_CO_OWNER_GROUP.

Reply:
do you mean that daos query command or the daos_cont_query() call? because i think the former is not really listing.

@wiliamhuang i do not want to ask for extra work for this PR. if the behavior before was to add the attribute even on an empty property (like the owner ones), then it's fine to keep it that way, and we can revisit that in the DM WG to see whether we can / want to change this.

Reply:
@mchaarawi I recalled why I chose writing empty property string instead of skipping the property completely.
In daos_cont_serialize_props(), the total number of properties is written into file at the very beginning. After that, individual property is written one by one. To skip the property with empty strings, we can do,
1) Scan all properties and adjust the total number of properties actually to save before writing the total number of properties.
or
2) Whenever we find a property we want to skip (e.g., empty string), we need to update the saved total number of properties in file.

Writing empty string seems the easiest way to me.

Reply:
The previous behavior on a NULL string was to return EINVAL. I agree that writing an empty string is the easiest. Otherwise, we could consider writing the number of properties after serializing the properties to avoid a double scan.

Reply:
yea it looks like something we can improve in general. don't worry about this for this PR then.